### PR TITLE
fix. URI path is empty

### PR DESCRIPTION
**Pull request by Instrye**
fix. #2822. path is empty and '', URI can't normal work

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/HTTP/URI.php [hunk 1]
```diff
@@ -726,7 +734,16 @@ public function refreshPath()
...
+		$tempPath = trim($this->path, '/');
+		if ($tempPath === '')
+		{
+			$this->segments = [];
+		}
+		else
+		{
+			$this->segments = explode('/', $tempPath);
+		}
+		$this->segments = explode('/', trim($this->path, '/'));
```
Comment (atishamte) [inline system/HTTP/URI.php:746]: conflicted line with above logic

File: system/HTTP/URI.php [hunk 2]
```diff
@@ -726,7 +734,16 @@ public function refreshPath()
 	{
 		$this->path = $this->filterPath(implode('/', $this->segments));
 
-		$this->segments = explode('/', $this->path);
+		$tempPath = trim($this->path, '/');
```
Comment (atishamte) [inline system/HTTP/URI.php:737]: Can we write this in ternary

File: system/HTTP/URI.php (hunk 1 above)
Comment (Instrye) [reply system/HTTP/URI.php:746]: thanks.

File: system/HTTP/URI.php (hunk 2 above)
Comment (Instrye) [reply system/HTTP/URI.php:737]: thanks.

**Pull request merged**